nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/RemoteProcessGroupResource.java

```diff
@@ -829,6 +833,105 @@ public Response updateRemoteProcessGroupRunStatus(
         );
     }
 
+    /**
+     * Updates the operational status for all remote process groups in the specified process group with the specified value.
+     *
+     * @param httpServletRequest                request
+     * @param processGroupId                    The id of the process group in which all remote process groups to update.
+     * @param requestRemotePortRunStatusEntity  A remotePortRunStatusEntity that holds the desired run status
+     * @return A response with an array of RemoteProcessGroupEntity objects.
+     */
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("process-group/run-status/{id}")
```

Comment:
Minor: the URI looks like the following: `http://localhost:8888/nifi-api/remote-process-groups/process-group/run-status/82ed213b-0178-1000-f0de-db399db203fc`. Is this intentional? As the method name differs from the original by using plural, I would suggest something like: `http://localhost:8888/nifi-api/remote-process-groups/run-statuses/82ed213b-0178-1000-f0de-db399db203fc`

Reply:
I can go with your consideration. LGTM

Reply:
The existing endpoint for starting/stopping all processors in a process group is:
`/flow/process-groups/{id}`
For enabling/disabling controller services in a process group:
`/flow/process-groups/{id}/controller-services`

Following the same scheme, I think the new endpoint should be:
`/flow/process-groups/{id}/remote-process-groups`

Reply:
The `RemoteProcessGroup` is significantly different from `Connectables` (handled by the `/flow/process-groups/{id}` endpoint) and the `ControllerServices` (handled by the `/flow/process-groups/{id}/remote-process-groups`) endpoint.

The most prominent manifestation of this is how the merging of the request fits perfectly into the `RemoteProcessGroupsEndpointMerger` and not at all into the `FlowMerger`.

There are other signs that this functionality is more suited to handled here, like private methods in this class that would be needed to be accessed from the `FlowResource`.

Leaving it here.

Reply:
I myself am okay with this but still not fund of the naming of the URI

Reply:
@tpalfy I can see the problematic points now (eg. private methods) and leaving the endoint in RemoteProcessGroup is fine with me.

Regarding the path, I would suggest a minor change: `remote-process-groups/process-group/{id}/run-status`
In case of other endpoints, the {id} follows the entity it identifies (in our case it is a process group id). So I believe it would be more straightforward in this way and also similar to the single RPG start/stop endpoint: `/remote-process-groups/{id}/run-status`